flink-runtime/src/main/java/org/apache/flink/runtime/rpc/RpcEndpoint.java

```diff
@@ -79,6 +79,9 @@
 	 * of the executing rpc server. */
 	private final MainThreadExecutor mainThreadExecutor;
 
+	/** Rpc endpoint's state. */
+	private State state;
```

Comment:
Maybe `volatile`?

Reply:
As the getter says the state is supposed to be read/written only in the main thread of the endpoint then there is no need for the synchronisation and I want to avoid confusion but adding volatile.

Reply:
Thanks for the clarification. I see the document below. We can also document it here or use a somewhat `@GuardBy` annotation. Although I'm not sure how to express guarded by MainThread with `@GuardBy`.

Reply:
Not sure we have something like that, I can only duplicate the comment from the getter here.
AFAIK, it is our usual implicit assumption about non-final variables in endpoints that they are confined to the main thread.